### PR TITLE
516 Recall Interactor

### PR DESCRIPTION
**Pull request by sadlerw**
* add new recalled status to case
* add set status to recalled to case
* add recall to workitem
* create interactor

### CODE REVIEW

File: shared/src/business/entities/WorkItem.js [hunk 1]
```diff
@@ -107,6 +107,21 @@ WorkItem.prototype.assignToIRSBatchSystem = function({ userId }) {
   );
 };
 
+WorkItem.prototype.recallFromIRSBatchSystem = function({ user }) {
+  this.assignToUser({
+    assigneeId: user.userId,
+    role: 'petitionsclerk',
+    assigneeName: user.name,
+  });
+  this.addMessage(
```
Comment (codyseibert) [inline shared/src/business/entities/WorkItem.js:117]: I think we also need to change the WorkItem's section back to 'petitions' at some point before we call saveCase, probably in this method.

File: shared/src/business/entities/WorkItem.js [hunk 2]
```diff
@@ -107,6 +107,21 @@ WorkItem.prototype.assignToIRSBatchSystem = function({ userId }) {
   );
 };
 
+WorkItem.prototype.recallFromIRSBatchSystem = function({ user }) {
+  this.assignToUser({
+    assigneeId: user.userId,
+    role: 'petitionsclerk',
```
Comment (kkoskelin) [inline shared/src/business/entities/WorkItem.js]: will this eventually become `user.role` ?

**Review by kkoskelin — APPROVED** [review]

**Review by kkoskelin — APPROVED** [review]

File: shared/src/business/entities/WorkItem.js (hunk 2 above)
Comment (sadlerw) [reply shared/src/business/entities/WorkItem.js]: no only the petitionsclerk role can perform this operation but I think that's better

**Review by codyseibert — APPROVED** [review]

File: shared/src/business/entities/WorkItem.js (hunk 1 above)
Comment (sadlerw) [reply shared/src/business/entities/WorkItem.js:117]: I'll add

Comment (sadlerw) [reply shared/src/business/entities/WorkItem.js:117]: oh you did - thank you!

**Pull request merged**